src/components/common/CheckWallet/index.tsx

```diff
@@ -28,28 +29,39 @@ const CheckWallet = ({
   allowNonOwner,
   noTooltip,
   checkNetwork = false,
+  allowUndeployedSafe = false,
 }: CheckWalletProps): ReactElement => {
   const wallet = useWallet()
   const isSafeOwner = useIsSafeOwner()
-  const isSpendingLimit = useIsOnlySpendingLimitBeneficiary()
+  const isOnlySpendingLimit = useIsOnlySpendingLimitBeneficiary()
   const connectWallet = useConnectWallet()
   const isWrongChain = useIsWrongChain()
   const isDelegate = useIsWalletDelegate()
 
   const { safe } = useSafeInfo()
 
-  const isCounterfactualMultiSig = !allowNonOwner && !safe.deployed && safe.threshold > 1
+  const isUndeployedSafe = !safe.deployed
 
-  const message =
-    wallet &&
-    (isSafeOwner || allowNonOwner || (isSpendingLimit && allowSpendingLimit) || isDelegate) &&
-    !isCounterfactualMultiSig
-      ? ''
-      : !wallet
-      ? Message.WalletNotConnected
-      : isCounterfactualMultiSig
-      ? Message.CounterfactualMultisig
-      : Message.NotSafeOwner
+  const message = useMemo(() => {
+    if (!wallet) {
+      return Message.WalletNotConnected
+    }
+    if ((!allowNonOwner && !isSafeOwner && !isDelegate) || (isOnlySpendingLimit && !allowSpendingLimit)) {
+      return Message.NotSafeOwner
+    }
+    if (isUndeployedSafe && !allowUndeployedSafe) {
+      return Message.SafeNotActivated
+    }
```

Comment:
I would reorder these two conditions to preserve the same order as previously since the check for NotSafeOwner is more specific than for SafeNotActivated. 

Do we need the empty string return still or does it also work with undefined?

Reply:
We do not need an empty string, it also works with undefined:
`  if (!message) return children(true)
`
I'll adjust the order 👍